x-pack/plugins/security_solution/public/management/components/console/mocks.tsx

```diff
@@ -119,32 +127,51 @@ export const getConsoleSelectorsAndActionMock = (
  * @param useKeyboard
  * @param dataTestSubj
  */
-export const enterConsoleCommand = (
+export const enterConsoleCommand = async (
   renderResult: ReturnType<AppContextTestRender['render']>,
+  user: UserEvent,
   cmd: string,
   {
     inputOnly = false,
     useKeyboard = false,
     dataTestSubj = 'test',
-  }: Partial<{ inputOnly: boolean; useKeyboard: boolean; dataTestSubj: string }> = {}
-): void => {
+    submitClick = false,
+  }: Partial<{
+    inputOnly: boolean;
+    useKeyboard: boolean;
+    dataTestSubj: string;
+    submitClick: boolean;
+  }> = {}
+): Promise<void> => {
   const keyCaptureInput = renderResult.getByTestId(`${dataTestSubj}-keyCapture-input`);
 
-  act(() => {
-    if (useKeyboard) {
-      userEvent.click(keyCaptureInput);
-      userEvent.keyboard(cmd);
-    } else {
-      userEvent.type(keyCaptureInput, cmd);
-    }
-
-    if (!inputOnly) {
-      userEvent.keyboard('{enter}');
-    }
-  });
+  if (keyCaptureInput === null) {
+    throw new Error(`No input found with test-subj: ${dataTestSubj}-keyCapture`);
+  }
+
+  if (useKeyboard) {
+    await user.click(keyCaptureInput);
+    await user.keyboard(cmd);
+  } else {
+    await user.type(keyCaptureInput, cmd);
+  }
+
+  // user-event v14 has a problem with {enter} not working on certain inputs
+  // so this provides a workaround to submit via click instead.
+  // See here for a related discussion: https://github.com/testing-library/user-event/discussions/1164
+  if (submitClick) {
+    await user.click(renderResult.getByTestId(`${dataTestSubj}-inputTextSubmitButton`));
+    return;
+  }
+
+  if (!inputOnly) {
+    await user.keyboard('{enter}');
+  }
 };
 
 export const getConsoleTestSetup = (): ConsoleTestSetup => {
+  // Workaround for timeout via https://github.com/testing-library/user-event/issues/833#issuecomment-1171452841
+  const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
```

Comment:
The recommended usage pattern in v14 is to use `const user = userEvent.setup()` once per test and then reuse `user` for that test. If you end up with multiple uses of `.setup()` within a test it will not work and events will not trigger and time out unfortunately. So we cannot simply import `userEvent` in another file and reuse it. 

The old pattern of just using `userEvent` directly from the import is still supported in v14 and will call `setup` on each use under the hood, but it's not recommended anymore. And this comes with a bit of a timing penalty and we cannot make use of the `advanceTimers` setting, that's what caused a lot of tests to time out unfortunately.

Reply:
Thanks for the context.